web/api/v1/api.go

```diff
@@ -1123,11 +1123,19 @@ func (api *API) remoteRead(w http.ResponseWriter, r *http.Request) {
 		for i, query := range req.Queries {
 			err := api.remoteReadQuery(ctx, query, externalLabels, func(querier storage.Querier, selectParams *storage.SelectParams, filteredMatchers []*labels.Matcher) error {
 				// The streaming API provides sorted series.
-				set, _, err := querier.SelectSorted(selectParams, filteredMatchers...)
+				set, ws, err := querier.SelectSorted(selectParams, filteredMatchers...)
 				if err != nil {
 					return err
 				}
 
+				if len(ws) > 0 {
+					msg := ""
+					for _, w := range ws {
+						msg += w.Error() + ";"
+					}
+					level.Warn(api.logger).Log("remote read warnings", "warnings", msg)
```

Comment:
This could be very spammy, this should not be in application logs.

Reply:
How would you handle this then?

Reply:
It's in the query logs if you really want that.

Reply:
kk, adding TODO